test-data/unit/fine-grained.test

```diff
@@ -4914,6 +5069,76 @@ class A: pass
 ==
 ==
 
+[case testChangeGenericFunctionToVariable]
+import a
+x: int
+y: int = a.f(x)
+class Dummy:
+    def g(self) -> None:
+        a.f(x)
+[file a.py]
+from typing import TypeVar
+T = TypeVar('T')
+def f(x: T) -> T:
+    pass
+[file a.py.2]
+from typing import TypeVar
+T = TypeVar('T')
+f = 42
+[out]
+==
+main:3: error: "int" not callable
+main:6: error: "int" not callable
+
+[case testChangeGenericClassToVariable]
+import a
+x: int
+a.A(x)
+class Dummy:
+    def g(self) -> None:
+        a.A(x)
+[file a.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+class A(Generic[T]):
+    def __init__(self, x: T) -> None:
+        pass
+[file a.py.2]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+A = 'no way'
+[out]
+==
+main:3: error: "str" not callable
+main:6: error: "str" not callable
+
+[case testChangeGenericMethodToVariable]
+import a
+x: int
+y: int = a.A(x).f()
+class Dummy:
+    def g(self) -> None:
+        a.A(x).f()
+[file a.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+class A(Generic[T]):
+    def __init__(self, x: T) -> None:
+        pass
+    def f(self) -> T:
+        pass
+[file a.py.2]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+class A(Generic[T]):
```

Comment:
Maybe also test changing a module to a non-module?

Reply:
Good suggestion! This uncovered one more bug.

Reply:
Hm, it looks like we need special handling of module aliases.

Reply:
OK, looks like it is a know issue https://github.com/python/mypy/issues/4532

Reply:
Hm, I added another test and found a different issue, will try to fix it.

Reply:
The solution is not very trivial and also solves #4532, so I will add these tests in a separate PR, together with a fix for them.